spinta/datasets/commands/check.py

```diff
@@ -1,14 +1,31 @@
+import re
+
 from spinta import commands
-from spinta.components import Context
+from spinta.components import Context, Config
 from spinta.datasets.components import Dataset, Resource
+from spinta.exceptions import InvalidName
 
 
 @commands.check.register()
 def check(context: Context, dataset: Dataset):
     for resource in dataset.resources.values():
         commands.check(context, resource)
 
+    config: Config = context.get('config')
+
+    if config.check_names:
+        name = dataset.name
+        is_lowercase = re.compile(r'^[a-z0-1_]*(/[a-z0-1_]*)+$')
+
+        # Dataset name check
+        if is_lowercase.match(name) is None or name.startswith('/') or name.endswith('/') or name.count('//'):
```

Comment:
All checks should be done only with regexes. There is no need to manually check slashes.